### ISSUE TITLE
Doing BREAKING changes in PATCH versions

### ISSUE THREAD

**Issue by montella1507** [telerik/kendo-themes#1257]
You have deleted component .scss files in patch version....

For example:
Kendo-theme-default  4.12.0 -> 4.12.2 removed ALL components scss files:
...
@import '~@progress/kendo-theme-default/scss/autocomplete.scss';
...

It is cool you added _index.scss however it is breaking change because it needs extensive configuration in webpack configuration etc. 

Please AVOID these breaking changes in patch versions, you destroying your paying customers because of ~ in package.json (all projects stoped working after reinstall of node modules) 

WORKAROUND:
Fix version to 4.12.0  because _index.scss resolution is not even supported in node-sass yet

**Comment by joneff** [comment]
@montella1507  that's the least breaking thing we came up with. While I understand why you may be frustrated, considering how huge change migration to sass module is, switch to index imports was a requirement we had to do from the start.

I should note that we didn't discard the normal files for import, nor we rely on index file resolution; instead we switch to index imports for future proofing the themes. To import autocomplete, per se, you should tweak your code like so:

```scss
// before
@import "~@progress/kendo-theme-default/scss/autocomplete";
// or
@import "~@progress/kendo-theme-default/scss/autocomplete.scss";

// now
@import "~@progress/kendo-theme-default/scss/autocomplete/_index.scss";
```

That being said, you do raise an interesting point: that should have been a minor version, not a patch. Looking at the commit bce25a70c766f2eca708af876331cf05d58f1f29, the commit message is with a `feat` prefix, which should have bumped the minor version, not the patch!

**Comment by montella1507** [comment]
> That being said, you do raise an interesting point: that should have been a minor version, not a patch. Looking at the commit bce25a7, the commit message is with a feat prefix, which should have bumped the minor version, not the patch!

Yes please, thats all i am talking about.  It breaks projects atfer installation because of semantic versioning and ^/~  version modifiers. 

I think these breaking changes should be in Major versions only - **however** it may be difficult in big solutions consisting of several packages (you will need to keep major versions in sync - like angular does).

But if you will do breaking changes (that breaks build) in MINOR releases, please DO NOT use ~   in base guides / tutorials in package.json. 

Developer can use ~ in version if it is safe (not breaking change in MINOR version).
So if you keep patch versions safe to update (no breaking changes) we can use ^ in packages.

**Comment by joneff** [comment]
That's the only major change (that requires tweaking) until v5, which is going not only to reply on index imports by default, but is going to take full advantage of [sass module system](https://sass-lang.com/blog/the-module-system-is-launched) and `@use`. v5 is going to be truly incompatible.

**Note**: before we bump to v5, we are going to ship as much as improvements / fixes as possible into v4! We are currently targeting v5 in about 20 or so months, around **October 2021**.

On the change log bit, I skimmed trough the change log and the `feat` isn't even there, so that's a bummer.

Lastly, after you tweak your code to import `component/_index.scss` instead of `component.scss`, [like we do in the themes](https://github.com/telerik/kendo-themes/blob/develop/packages/default/scss/all.scss), does your build pass? Locally, we use `node-sass` for the builds (and we'll continue to do so until v5) and our builds are passing.

**Comment by montella1507** [comment]
Yes, when i change imports like this:
![image](https://user-images.githubusercontent.com/10418323/74648672-6f1b2300-517e-11ea-8d76-c51849ad0045.png)


i am able to build. however, there are still some breaking changes (different margins etc)  so we keep fixed to 4.12.0 atm. 

For example, this (kendo-floating-label):
4.12.2
![image](https://user-images.githubusercontent.com/10418323/74648809-aa1d5680-517e-11ea-8191-083879ae7bf5.png)

4.12.0
![image](https://user-images.githubusercontent.com/10418323/74648902-cf11c980-517e-11ea-9aa3-cf5144a67750.png)


Width is different (100% in 4.12.0   , 12.4em in 4.12.2)

**Comment by joneff** [comment]
Regarding the differences:

* precision for padding top looks like a setting
* we haven't actually changed 12.4 em to 100%

Looking at the source for theme default, the width maps to the same variable it has for while now:

https://github.com/telerik/kendo-themes/blob/f747e59f1fb33cb23eeb282346620482ee051e33/packages/default/scss/input/_layout.scss#L98-L106

Looking at the variables file, that variable is still 12.4em:

https://github.com/telerik/kendo-themes/blob/f747e59f1fb33cb23eeb282346620482ee051e33/packages/default/scss/_variables.scss#L662

Looking at the [dist](https://unpkg.com/@progress/kendo-theme-default@4.12.2/dist/all.css), the value is 12.4em.

Would it be possible that you override it somewhere in your code?
